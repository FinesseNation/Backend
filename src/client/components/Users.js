import React, {Component} from 'react';
import PropTypes from 'prop-types';
import queryString from 'query-string';
import axios from 'axios';

class Users extends Component {
    constructor(props) {
        super(props);
        this.state = {
            email: "",
            password: "",
            confirmPassword: "",
            validRequest: false
        };
    }

    componentDidMount() {
        const params = queryString.parse(location.search);
        const email = params["email"];
        const token = params["token"];

        axios.post('api/user/checkEmailTokenExists', {
            emailId: email,
            token: token
        },{
            headers: {
                "api_token": process.env.API_TOKEN
            }
        }).then((res) => {
            if (res.status === 200) {
                console.log("Rendering password reset form");
                this.setState({email: email});
                this.setState({validRequest: true});
            } else {
                console.log("Rendering invalid password reset request page");
            }
        }).catch(err => console.log(err));
    }

    changePassword = (email, password, confirmPassword) => {
        if (password !== confirmPassword) {
            alert("Passwords do not match!");
        } else {
            axios.post('api/user/changePassword', {
                emailId: email,
                password: confirmPassword
            },{
                headers: {
                    "api_token": process.env.API_TOKEN
                }
            }).then((res) => {
                if (res.status === 200) {
                    alert("Password successfully updated");
                } else {
                    alert("Error - unable to update password");
                }
            }).catch(err => console.log(err));
        }
    };

    render() {
        if (this.state.validRequest) {
            return (
                <div className="container">
                    <h3>Reset Password for User (email = {this.state.email})</h3>
                    <form onSubmit={() => this.changePassword(this.state.email, this.state.password, this.state.confirmPassword)}>
                        <label>
                            New Password:
                            <input type="password" onChange={(i) => this.setState({password: i.target.value})}/>
                        </label>
                        <label>
                            Retype New Password:
                            <input type="password" onChange={(i) => this.setState({confirmPassword: i.target.value})}/>
                        </label>
                        <button type="submit" className="btn btn-default">Submit</button>
                    </form>
                </div>
            );
        } else {
            return (
                <div className="container">
                    <h3>Email/Token invalid or token has expired. Please retry password reset request.</h3>
                </div>
            );
        }
    }
}

Users.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default Users;
