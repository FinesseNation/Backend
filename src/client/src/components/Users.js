import React, {Component} from 'react';
import queryString from 'query-string';
import axios from 'axios';

class Users extends Component {
    constructor() {
        super();
        this.state = {
            email: "",
            password: "",
            confirmPassword: "",
            validRequest: false,
            passwordChanged: false,
        };
        this.changePassword = this.changePassword.bind(this);
    }

    componentDidMount() {
        const params = queryString.parse(window.location.search);
        const email = params["email"];
        const token = params["token"];

        if (email && token) {
            axios.post('api/user/checkEmailTokenExists', {
                emailId: email,
                token: token
            },{
                headers: {
                    "api_token": process.env.REACT_APP_API_TOKEN
                }
            }).then((res) => {
                if (res.status === 200) {
                    console.log("Rendering password reset form");
                    this.setState({email: email});
                    this.setState({validRequest: true});
                } else {
                    console.log("Rendering invalid password reset request page");
                }
            }).catch(err => console.log(err));
        }
    }

    changePassword(event) {
        event.preventDefault();
        if (this.state.password !== this.state.confirmPassword) {
            alert("Passwords do not match!");
        } else {
            axios.post('api/user/changePassword', {
                emailId: this.state.email,
                password: this.state.confirmPassword
            },{
                headers: {
                    "api_token": process.env.REACT_APP_API_TOKEN
                }
            }).then((res) => {
                if (res.status === 200) {
                    this.setState({validRequest: false});
                    this.setState({passwordChanged: true});
                    alert("Password successfully updated");
                } else {
                    alert("Error - unable to update password");
                    this.setState({validRequest: false});
                }
            }).catch(err => console.log(err));
        }
    };

    render() {
        if (this.state.passwordChanged) {
            return (
                <div className="container">
                    <h3>Password successfully changed for user ({this.state.email}).</h3>
                </div>
            )
        }
        else if (this.state.validRequest) {
            return (
                <div className="container">
                    <h3>Reset Password for User ({this.state.email}).</h3>
                    <form onSubmit={this.changePassword}>
                        <label>
                            New Password:
                            <input type="password" onChange={(i) => this.setState({password: i.target.value})}/>
                        </label>
                        <label>
                            Retype New Password:
                            <input type="password" onChange={(i) => this.setState({confirmPassword: i.target.value})}/>
                        </label>
                        <button type="submit" className="btn btn-default">Submit</button>
                    </form>
                </div>
            );
        } else {
            return (
                <div className="container">
                    <h3>Error: Email/Token invalid or token has expired. Please retry password reset request.</h3>
                </div>
            );
        }
    }
}

export default Users;
