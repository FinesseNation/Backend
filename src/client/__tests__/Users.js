import React from 'react';
import renderer from 'react-test-renderer';
import Users from '../components/Users';

test('User password reset page should load', () => {
    const component = renderer.create(
        <Users/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
