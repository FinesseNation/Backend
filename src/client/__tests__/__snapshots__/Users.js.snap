// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User password reset page should load 1`] = `
<div
  className="container"
>
  <h3>
    Email/Token invalid or token has expired. Please retry password reset request.
  </h3>
</div>
`;
